app/models/manageiq/providers/vmware/manager_auth_mixin.rb

```diff
@@ -36,10 +44,23 @@ def connect(options = {})
     self.class.raw_connect(server, port, username, password)
   end
 
+  def translate_exception(err)
+    case err
+      when Fog::Compute::VcloudDirector::Unauthorized
+        MiqException::MiqInvalidCredentialsError.new "Login failed due to a bad username or password."
+      when Excon::Errors::Timeout
+        MiqException::MiqUnreachableError.new "Login attempt timed out"
+      when Excon::Errors::SocketError
+        MiqException::MiqHostError.new "Socket error: #{err.message}"
+      when MiqException::MiqInvalidCredentialsError, MiqException::MiqHostError # from amqp
```

Comment:
Yes, they can only come from AMQP since DEFAULT raises Fog exceptions directly (which is actually the very reason why `require 'fog/vcloud_director'` at the top of the file is needed). But I've removed the comment anyway since in the future perhaps METERING will wrapp its errors in this generic form as well.